app/CreateDeal.tsx

```diff
@@ -17,9 +26,10 @@ export function CreateDeal(props: {
   const [errorMessage, setErrorMessage] = useState<string>("");
   const [itemName, setItemName] = useState<string>("");
   const [pricePerUnit, setPricePerUnit] = useState<string>("0");
-  const [date, setDate] = useState(new Date());
+  const [selectedUnit, setSelectedUnit] = useState<string>("");
```

Comment:
the selected unit should be of type `Units | undefined`, not string, otherwise it's just as loosey-goosey as before.
when selecting from a set list, we shouldn't use empty string as a default since it's not one of the available items in the list, have it be explicitly undefined or null instead

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/30/commits/b60205858cdc91a62b2d292d032da4995b2a88b4